### PR TITLE
Add basic find and edit features for Module Manager

### PR DESCRIPTION
**Pull request by mathanmahe**

### CODE REVIEW

File: src/main/java/seedu/allonus/modules/StudyManager.java [hunk 1]
```diff
@@ -106,6 +110,78 @@ public void deleteModule(String userInput) {
...
+                        editModuleDay(moduleToEdit, editUserInput);
+                        printMessage(moduleToEdit.toString());
+                    } else if (editUserInput.startsWith("time")) {
+                        editModuleTime(moduleToEdit, editUserInput);
+                        printMessage(moduleToEdit.toString());
+                    } else {
+                        printMessage("Your Module was successfully edited! Here are the changes");
+                        printMessage(moduleToEdit.toString());
+                        isEditFinished = true;
+                    }
```
Comment (wli-linda) [inline src/main/java/seedu/allonus/modules/StudyManager.java:142]: Is it possible to extract this portion as a method? This seems a little deeply nested.

File: src/main/java/seedu/allonus/modules/StudyManager.java [hunk 2]
```diff
@@ -221,29 +321,34 @@ private String moduleCategoryChecker(String[] parameters) throws ModuleCategoryE
...
-                case "exam":
-                    category = "Exam";
-                    break;
-                default:
-                    printMessage("Category has to be one of lec,tut or exam");
-                    throw new ModuleCategoryException();
-                }
-                assert (category == "Lecture" || category == "Tutorial" || category == "Exam") : "category is not one"
-                        + " of lec, tut or exam";
+                category = validateModuleCategory(category);
```
Comment (wli-linda) [inline src/main/java/seedu/allonus/modules/StudyManager.java:324]: Nice extraction of methods :)

**Review by wli-linda — COMMENTED** [review]
Looks pretty good

File: src/main/java/seedu/allonus/modules/StudyManager.java [hunk 3]
```diff
@@ -106,6 +110,77 @@ public void deleteModule(String userInput) {
...
 
+    public void editModule(String userInput, TextUi ui) {
+        try {
+            String moduleIndexString = userInput.replace("edit ", "");
+            int moduleIndex = Integer.parseInt(moduleIndexString) - 1;
+            if (modulesList.get(moduleIndex) != null) {
+                Module moduleToEdit = modulesList.get(moduleIndex);
+                printMessage("Here is the module that you have chosen to edit:");
+                printMessage(moduleToEdit.toString());
+                printMessage("Choose the part that you would like to edit: ");
```
Comment (YitHien) [inline src/main/java/seedu/allonus/modules/StudyManager.java:121]: You could consider refactoring out the string constants here! And likewise for the subsequent strings too

File: src/main/java/seedu/allonus/modules/StudyManager.java [hunk 4]
```diff
@@ -168,6 +243,30 @@ public Module addModuleParser(String userInput) {
...
+        if (matches.size() == 0) {
+            printMessage("There are no modules that match \"" + moduleKeyword + "\"");
+        } else {
+            listMatches(matches);
+        }
+    }
+
+    private void listMatches(ArrayList<Module> matches) {
+        System.out.println("    Here are the matching tasks in your list:");
+        int i=1;
```
Comment (YitHien) [inline src/main/java/seedu/allonus/modules/StudyManager.java]: Minor coding standard violation here; consider adding some spaces surrounding the = sign

File: src/main/java/seedu/allonus/modules/StudyManager.java [hunk 5]
```diff
@@ -168,6 +243,30 @@ public Module addModuleParser(String userInput) {
         }
     }
 
+    public void findModule(String userInput) {
+        String moduleKeyword = userInput.replace("find ","");
+        ArrayList<Module> matches = new ArrayList<>();
+        for (Module m: modulesList) {
+            if (m.toString().contains(moduleKeyword)) {
```
Comment (YitHien) [inline src/main/java/seedu/allonus/modules/StudyManager.java:251]: If I'm not wrong, the user would be able to enter in a blank keyword to look for and the method may return every item in the list. Perhaps consider edge cases of inappropriate keywords for subsequent commits!

**Review by YitHien — COMMENTED** [review]
Except for a few minor edits, looks good to merge!

**Comment by mathanmahe** [comment]
@wli-linda, @YitHien Thank you for the code review. Will update these suggestions accordingly.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -8,6 +8,17 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Study Manager component
+API: `StudyManager.java`
+
+![](images/StudyManagerClass.png)
```
Comment (OzairHasan) [inline docs/DeveloperGuide.md:14]: I like the simplicity and clarity of the Class diagram.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -8,6 +8,17 @@
...
+
+The `StudyManager` component,
+1. Stores the academic schedule, i.e. all `Module` objects are contained in a `modulesList` object.
+2. The storage can be bound to `modulesList` list such that everytime a change is observed in the list it is saved on to a file.
+3. Does not depend on other components like `ExpenseTracker` and `ContactsManager`
+
```
Comment (OzairHasan) [inline docs/DeveloperGuide.md:20]: Perhaps a sequence diagram could be included to further elaborate on the operational design of the StudyManager component.

**Review by OzairHasan — COMMENTED** [review]
LGTM, further updates required in future iterations.

**Pull request merged**